Add megafauna variant of the big parrot; various tweaks/cleanup

babydoll opened the pull request:
:cl:
tweak: Bog-standard parrots will now retaliate briefly when attacked.
tweak: The space parrot occupying its temple in the asteroid will now retaliate with force when attacked. Beware!
rscadd: A giant parrot can now spawn as the grass planet's megafauna.
/:cl:

retypes parrot to retaliate
adds `simple_parrot` which bypasses a bunch of the parrot mechanics

all giant parrots have two abilities: they can knock you over with their big wings, or pry your helmet off

babydoll commented:
Abilities seem to work as intended

afterthought2 commented on code/modules/mob/living/simple_animal/hostile/retaliate/giant_parrot.dm:
```diff
@@ -0,0 +1,88 @@
+		if(prob(90))
+			H.Weaken(rand(2,3))
+			last_ability = world.time + ability_cooldown
+			visible_message(SPAN_MFAUNA("\The flaps its wings mightily and bowls over \the [H]!"))
```
`\The [src]`

afterthought2 commented on code/modules/mob/living/simple_animal/hostile/retaliate/giant_parrot.dm:
```diff
@@ -0,0 +1,88 @@
+			last_ability = world.time + ability_cooldown
+			visible_message(SPAN_MFAUNA("\The flaps its wings mightily and bowls over \the [H]!"))
+
+		else if(H.head && istype(H.head, /obj/item/clothing/head/helmet))
```
Should this also put the abilities on cooldown?

afterthought2 commented on code/modules/mob/living/simple_animal/hostile/retaliate/giant_parrot.dm:
```diff
@@ -0,0 +1,88 @@
+				H.unEquip(helm, src)
+				if(helm.loc == src)
+					held_item = helm
+					drop_held_item()
```
May be simpler to just `H.unEquip(helm, get_turf(src))`, unless dropping does something special. If you do it the current way, you also have to check that you aren't already holding an item.

babydoll replied:
I thought that drop_held_item might display a message but it doesn't, forgot to change this back

babydoll commented:
ready

afterthought2 commented:
Oh, rip, conflicts with mice, @babydoll.

PsiOmegaDelta commented on code/modules/mob/living/simple_animal/hostile/retaliate/giant_parrot.dm:
```diff
@@ -0,0 +1,106 @@
+				if("red")
+					SetName("phoenix")
+				if("black")
+					SetName("lord of birds")
```
This makes it less of a pain if someone ever wants to subtype a `/mob/living/simple_animal/hostile/retaliate/parrot/space` because the randomness is easily adjusted or disabled:
```
/decl/parrot_subspecies
    var/name = "giant parrot"
    var/icon_set = "parrot"

[etc]

/mob/living/simple_animal/hostile/retaliate/parrot/space
    var/list/sub_species

/mob/living/simple_animal/hostile/retaliate/parrot/space/Initialize()
    . = ..()
    var/subspecies_type = safepick(sub_species)
    if(subspecies_type)
        var/decl/parrot_subspecies/ps = decls_repository.get_decl(subspecies_type)
        SetName(ps.name)
        icon_set = ps.icon_set
        update_icon

/mob/living/simple_animal/hostile/retaliate/parrot/space/random
    sub_species = list(/datum/parrot_subspecies, etc.)
```

PsiOmegaDelta commented on code/modules/mob/living/simple_animal/hostile/retaliate/giant_parrot.dm:
```diff
@@ -0,0 +1,106 @@
+			last_ability = world.time + ability_cooldown
+			visible_message(SPAN_MFAUNA("\The [src] flaps its wings mightily and bowls over \the [H] with a gust!"))
+
+		else if(H.head && istype(H.head, /obj/item/clothing/head/helmet))
```
Should really use `H.get_equipped_item(slot_head)`

PsiOmegaDelta commented on code/modules/mob/living/simple_animal/hostile/retaliate/giant_parrot.dm:
```diff
@@ -0,0 +1,106 @@
+
+		else if(H.head && istype(H.head, /obj/item/clothing/head/helmet))
+			var/obj/item/clothing/head/helmet/helm = H.head
+			if(istype(helm, /obj/item/clothing/head/helmet/space/rig) && istype(H.back, /obj/item/weapon/rig))
```
Not sure I like these snowflake checks

babydoll replied:
What do you think could be done instead?

comma replied:
can check `canremove`, rigs set it to 0 on their pieces

babydoll replied:
I'd still need to check if it's a rig helmet to print the correct message

babydoll committed:
Add megafauna variant of the giant parrot

Move simple animal icons to their own subfolder
Clean up simple animal vars
Set up giant parrot subspecies

babydoll commented:
Squashed
